interactions/models/internal/command.py

```diff
@@ -195,7 +198,11 @@ def param_config(self, annotation: Any, name: str) -> Tuple[Callable, Optional[d
         return (annotation, getattr(annotation, name, None))
 
     async def call_callback(self, callback: Callable, context: "BaseContext") -> None:
-        await self.call_with_binding(callback, context, **context.kwargs)  # type: ignore
+        val = await self.call_with_binding(callback, context, **context.kwargs)  # type: ignore
+        if isinstance(val, str) and isinstance(context, SendMixin):
+            await context.send(val)
+        elif isinstance(val, Embed) and isinstance(context, SendMixin):
+            await context.send(embed=val)
```

Comment:
Haven't tested this, but this brings forth the proposal in suggestion form.

```suggestion
            await context.send(embed=val)
        elif all(isinstance(val, list[Embed])) and isinstance(context, SendMixin):
            await context.send(embeds=[embed for embed in val])
        elif isinstance(val, File) and isinstance(context, SendMixin):
            await context.send(file=val)
        elif all(isinstance(val, list[File])) and isinstance(context, SendMixin):
            await context.send(files=[file for file in val])
```

Reply:
Fun fact: `isinstance(..., list[T])` doesn't do what you think it does. It just checks that the object is a list, and ignores the generic parameter entirely.  

There are a couple bigger changes I need to make to the PR before it lands, and I'm hoping to have time for them this weekend